### ISSUE TITLE
KeyError: '__dummy__'  for pd.crosstab in pandas

### ISSUE THREAD

**Issue by songhuiming** [pydata/pandas#10291]
get ~~ KeyError: '__dummy__' ~~ when I run the following:

    np.random.seed(seed = 99)
    s = np.random.randint(1,10,200)
    s = pd.Series(np.where(s > 9, np.nan, s))
    s1 = s[:100]
    s2 = s[100:]
    pd.crosstab(s1, s2)

    KeyError: '__dummy__

**Comment by lexual** [comment]
Even simpler example. Perhaps something to do with the indices not overlapping at all.


    s1 = pd.Series([1, 2, 3], index=[1, 2, 3])
    s2 = pd.Series([4, 5, 6], index=[4, 5, 6])
    pd.crosstab(s1, s2)

**Comment by lexual** [comment]
Believe this is the root cause of things:

http://pandas.pydata.org/pandas-docs/stable/groupby.html#na-group-handling

**Comment by lexual** [comment]
Yes http://pandas.pydata.org/pandas-docs/stable/groupby.html#na-group-handling is this cause.

Because the 2 indices have no overlapping indexes, this means that each groupby ends up including a nan which then excludes it from groupby result.

You then end up with an empty dataframe and that is the cause of the  KeyError, as you're accessing df['__dummy__'] on an empty dataframe.

**Comment by jreback** [comment]
yeh, this should just be an empty frame, as there are no cross-tabulations.

**Comment by lexual** [comment]
So this is not a bug?

should we:
* raise exception
* return an empty dataframe?

**Comment by jreback** [comment]
return an empty frame

**Comment by dan7davis** [comment]
I'm getting the same KeyError: '__dummy__' for my grouped data.

And I'm not really sure how to fix it / what you mean by 'return an empty frame.' Care to dumb it down/show precisely what you mean?

Thanks!

**Comment by jreback** [comment]
@dan7davis this needs a fix that would return an empty frame when catching the ``KeyError`` exception raised by the example above

https://github.com/pydata/pandas/blob/master/pandas/tools/pivot.py#L151, just need something like:

```
try:
    table = table[values[0]]
except KeyError:
    pass
```

**Comment by dan7davis** [comment]
@jreback problem solved. thank you! really appreciate the alacrity

**Comment by jreback** [comment]
want to do a pull request to fix in master?

**Comment by dan7davis** [comment]
I'm (very) new to coding/python/GitHub, so unfortunately I have no idea
what that means. But it sounds useful for me to know & helpful for others,
so I'd be happy to learn/try..

**Comment by jreback** [comment]
contributing is a great way to learn ...., see our docs: http://pandas.pydata.org/pandas-docs/stable/contributing.html

any questions, pls ask.
